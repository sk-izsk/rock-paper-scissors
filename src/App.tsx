import React from 'react';
import { createUseStyles } from 'react-jss';
import 'ui-neumorphism/dist/index.css';
import { GameScreen } from './screens';
import { CustomTheme } from './theme';

const useStyles = createUseStyles((theme: CustomTheme) => ({
  '@global': {
    body: {
      padding: 0,
      margin: 0,
      backgroundColor: theme.colors.bgColor,
      fontFamily: theme.fontProperties.fontFamily,
      scrollBehavior: 'smooth',
    },
  },
}));

const App: React.FC = () => {
  useStyles();
  return (
    <>
      <GameScreen />
    </>
  );
};

export { App };
