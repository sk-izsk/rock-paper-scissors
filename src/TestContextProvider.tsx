import React from 'react';
import { ThemeProvider } from 'react-jss';
import { theme } from './theme';

const TestContextProvider: React.FC = ({ children }) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
};

export { TestContextProvider };
