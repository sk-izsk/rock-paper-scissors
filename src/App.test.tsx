import { render } from '@testing-library/react';
import React from 'react';
import { App } from './App';
import { TestContextProvider } from './TestContextProvider';

test('App renders correctly', () => {
  const { asFragment } = render(
    <TestContextProvider>
      <App />
    </TestContextProvider>,
  );
  expect(asFragment()).toMatchSnapshot();
});
