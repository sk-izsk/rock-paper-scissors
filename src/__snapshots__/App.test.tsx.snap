// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<DocumentFragment>
  <div
    class="mainContainer-0-2-1"
  >
    <div
      class="mainContainer-0-2-2"
    >
      <h5
        class="_3yPwv _2TwOi _11JjB _1uvOH header-0-2-6"
      >
        Mode : 
        <span
          class="greenText-0-2-8"
        >
          Normal
        </span>
      </h5>
      <div
        class="playersContainer-0-2-3"
      >
        <div
          class="_28vIk _2bGnb _2Xmdp _1SFBz card-0-2-15 cardContainer-0-2-5"
        >
          <div
            class="_2So3v _2uw1B "
          >
            <svg
              color="#2196f3"
              fill="currentColor"
              height="200"
              size="200"
              stroke="currentColor"
              stroke-width="0"
              style="color: rgb(33, 150, 243);"
              viewBox="0 0 512 512"
              width="200"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M408.864 79.052c-22.401-33.898-66.108-42.273-98.813-23.588-29.474-31.469-79.145-31.093-108.334-.022-47.16-27.02-108.71 5.055-110.671 60.806C44.846 105.407 0 140.001 0 187.429v56.953c0 32.741 14.28 63.954 39.18 85.634l97.71 85.081c4.252 3.702 3.11 5.573 3.11 32.903 0 17.673 14.327 32 32 32h252c17.673 0 32-14.327 32-32 0-23.513-1.015-30.745 3.982-42.37l42.835-99.656c6.094-14.177 9.183-29.172 9.183-44.568V146.963c0-52.839-54.314-88.662-103.136-67.911zM464 261.406a64.505 64.505 0 0 1-5.282 25.613l-42.835 99.655c-5.23 12.171-7.883 25.04-7.883 38.25V432H188v-10.286c0-16.37-7.14-31.977-19.59-42.817l-97.71-85.08C56.274 281.255 48 263.236 48 244.381v-56.953c0-33.208 52-33.537 52 .677v41.228a16 16 0 0 0 5.493 12.067l7 6.095A16 16 0 0 0 139 235.429V118.857c0-33.097 52-33.725 52 .677v26.751c0 8.836 7.164 16 16 16h7c8.836 0 16-7.164 16-16v-41.143c0-33.134 52-33.675 52 .677v40.466c0 8.836 7.163 16 16 16h7c8.837 0 16-7.164 16-16v-27.429c0-33.03 52-33.78 52 .677v26.751c0 8.836 7.163 16 16 16h7c8.837 0 16-7.164 16-16 0-33.146 52-33.613 52 .677v114.445z"
              />
            </svg>
            <h6
              class="_3yPwv _2wZct _11JjB _1uvOH "
            >
               : Rock
            </h6>
          </div>
        </div>
        <div
          class="_28vIk _2bGnb _2Xmdp _1SFBz card-0-2-15 cardContainer-0-2-5"
        >
          <div
            class="_2So3v _2uw1B "
          >
            <svg
              color="#2196f3"
              fill="currentColor"
              height="200"
              size="200"
              stroke="currentColor"
              stroke-width="0"
              style="color: rgb(33, 150, 243);"
              viewBox="0 0 512 512"
              width="200"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M408.864 79.052c-22.401-33.898-66.108-42.273-98.813-23.588-29.474-31.469-79.145-31.093-108.334-.022-47.16-27.02-108.71 5.055-110.671 60.806C44.846 105.407 0 140.001 0 187.429v56.953c0 32.741 14.28 63.954 39.18 85.634l97.71 85.081c4.252 3.702 3.11 5.573 3.11 32.903 0 17.673 14.327 32 32 32h252c17.673 0 32-14.327 32-32 0-23.513-1.015-30.745 3.982-42.37l42.835-99.656c6.094-14.177 9.183-29.172 9.183-44.568V146.963c0-52.839-54.314-88.662-103.136-67.911zM464 261.406a64.505 64.505 0 0 1-5.282 25.613l-42.835 99.655c-5.23 12.171-7.883 25.04-7.883 38.25V432H188v-10.286c0-16.37-7.14-31.977-19.59-42.817l-97.71-85.08C56.274 281.255 48 263.236 48 244.381v-56.953c0-33.208 52-33.537 52 .677v41.228a16 16 0 0 0 5.493 12.067l7 6.095A16 16 0 0 0 139 235.429V118.857c0-33.097 52-33.725 52 .677v26.751c0 8.836 7.164 16 16 16h7c8.836 0 16-7.164 16-16v-41.143c0-33.134 52-33.675 52 .677v40.466c0 8.836 7.163 16 16 16h7c8.837 0 16-7.164 16-16v-27.429c0-33.03 52-33.78 52 .677v26.751c0 8.836 7.163 16 16 16h7c8.837 0 16-7.164 16-16 0-33.146 52-33.613 52 .677v114.445z"
              />
            </svg>
            <h6
              class="_3yPwv _2wZct _11JjB _1uvOH "
            >
              Computer : Rock
            </h6>
          </div>
        </div>
      </div>
      <div
        class="mainContainer-0-2-16"
      >
        <div
          class="_cL4RU _17_OS  _3W7Om _CWiZp _39_ND _z9EYS button-0-2-17"
        >
          <button
            class="_3lLt8"
          >
            Rules
          </button>
        </div>
        <div
          class="_cL4RU _17_OS  _3W7Om _CWiZp _39_ND _z9EYS button-0-2-17"
        >
          <button
            class="_3lLt8"
          >
            <svg
              class="icon-0-2-19"
              color="#2196f3"
              fill="currentColor"
              height="20"
              size="20"
              stroke="currentColor"
              stroke-width="0"
              style="color: rgb(33, 150, 243);"
              viewBox="0 0 512 512"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M408.864 79.052c-22.401-33.898-66.108-42.273-98.813-23.588-29.474-31.469-79.145-31.093-108.334-.022-47.16-27.02-108.71 5.055-110.671 60.806C44.846 105.407 0 140.001 0 187.429v56.953c0 32.741 14.28 63.954 39.18 85.634l97.71 85.081c4.252 3.702 3.11 5.573 3.11 32.903 0 17.673 14.327 32 32 32h252c17.673 0 32-14.327 32-32 0-23.513-1.015-30.745 3.982-42.37l42.835-99.656c6.094-14.177 9.183-29.172 9.183-44.568V146.963c0-52.839-54.314-88.662-103.136-67.911zM464 261.406a64.505 64.505 0 0 1-5.282 25.613l-42.835 99.655c-5.23 12.171-7.883 25.04-7.883 38.25V432H188v-10.286c0-16.37-7.14-31.977-19.59-42.817l-97.71-85.08C56.274 281.255 48 263.236 48 244.381v-56.953c0-33.208 52-33.537 52 .677v41.228a16 16 0 0 0 5.493 12.067l7 6.095A16 16 0 0 0 139 235.429V118.857c0-33.097 52-33.725 52 .677v26.751c0 8.836 7.164 16 16 16h7c8.836 0 16-7.164 16-16v-41.143c0-33.134 52-33.675 52 .677v40.466c0 8.836 7.163 16 16 16h7c8.837 0 16-7.164 16-16v-27.429c0-33.03 52-33.78 52 .677v26.751c0 8.836 7.163 16 16 16h7c8.837 0 16-7.164 16-16 0-33.146 52-33.613 52 .677v114.445z"
              />
            </svg>
            Rock
          </button>
        </div>
        <div
          class="_cL4RU _17_OS  _3W7Om _CWiZp _39_ND _z9EYS button-0-2-17"
        >
          <button
            class="_3lLt8"
          >
            <svg
              class="icon-0-2-19"
              color="#2196f3"
              fill="currentColor"
              height="20"
              size="20"
              stroke="currentColor"
              stroke-width="0"
              style="color: rgb(33, 150, 243);"
              viewBox="0 0 448 512"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M372.57 112.641v-10.825c0-43.612-40.52-76.691-83.039-65.546-25.629-49.5-94.09-47.45-117.982.747C130.269 26.456 89.144 57.945 89.144 102v126.13c-19.953-7.427-43.308-5.068-62.083 8.871-29.355 21.796-35.794 63.333-14.55 93.153L132.48 498.569a32 32 0 0 0 26.062 13.432h222.897c14.904 0 27.835-10.289 31.182-24.813l30.184-130.958A203.637 203.637 0 0 0 448 310.564V179c0-40.62-35.523-71.992-75.43-66.359zm27.427 197.922c0 11.731-1.334 23.469-3.965 34.886L368.707 464h-201.92L51.591 302.303c-14.439-20.27 15.023-42.776 29.394-22.605l27.128 38.079c8.995 12.626 29.031 6.287 29.031-9.283V102c0-25.645 36.571-24.81 36.571.691V256c0 8.837 7.163 16 16 16h6.856c8.837 0 16-7.163 16-16V67c0-25.663 36.571-24.81 36.571.691V256c0 8.837 7.163 16 16 16h6.856c8.837 0 16-7.163 16-16V101.125c0-25.672 36.57-24.81 36.57.691V256c0 8.837 7.163 16 16 16h6.857c8.837 0 16-7.163 16-16v-76.309c0-26.242 36.57-25.64 36.57-.691v131.563z"
              />
            </svg>
            Paper
          </button>
        </div>
        <div
          class="_cL4RU _17_OS  _3W7Om _CWiZp _39_ND _z9EYS button-0-2-17"
        >
          <button
            class="_3lLt8"
          >
            <svg
              class="icon-0-2-19"
              color="#2196f3"
              fill="currentColor"
              height="20"
              size="20"
              stroke="currentColor"
              stroke-width="0"
              style="color: rgb(33, 150, 243);"
              viewBox="0 0 512 512"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M256 480l70-.013c5.114 0 10.231-.583 15.203-1.729l118.999-27.427C490.56 443.835 512 417.02 512 386.277V180.575c0-23.845-13.03-45.951-34.005-57.69l-97.999-54.853c-34.409-19.261-67.263-5.824-92.218 24.733L142.85 37.008c-37.887-14.579-80.612 3.727-95.642 41.201-15.098 37.642 3.635 80.37 41.942 95.112L168 192l-94-9.141c-40.804 0-74 32.811-74 73.14 0 40.33 33.196 73.141 74 73.141h87.635c-3.675 26.245 8.692 51.297 30.341 65.006C178.657 436.737 211.044 480 256 480zm0-48.013c-25.16 0-25.12-36.567 0-36.567 8.837 0 16-7.163 16-16v-6.856c0-8.837-7.163-16-16-16h-28c-25.159 0-25.122-36.567 0-36.567h28c8.837 0 16-7.163 16-16v-6.856c0-8.837-7.163-16-16-16H74c-34.43 0-34.375-50.281 0-50.281h182c8.837 0 16-7.163 16-16v-11.632a16 16 0 0 0-10.254-14.933L106.389 128.51c-31.552-12.14-13.432-59.283 19.222-46.717l166.549 64.091a16.001 16.001 0 0 0 18.139-4.812l21.764-26.647c5.82-7.127 16.348-9.064 24.488-4.508l98 54.854c5.828 3.263 9.449 9.318 9.449 15.805v205.701c0 8.491-5.994 15.804-14.576 17.782l-119.001 27.427a19.743 19.743 0 0 1-4.423.502h-70z"
              />
            </svg>
            Scissors
          </button>
        </div>
        <div
          class="_cL4RU _17_OS  _3W7Om _CWiZp _39_ND _z9EYS button-0-2-17"
        >
          <button
            class="_3lLt8"
          >
            <svg
              class="icon-0-2-19"
              color="#2196f3"
              fill="currentColor"
              height="20"
              size="20"
              stroke="currentColor"
              stroke-width="0"
              style="color: rgb(33, 150, 243);"
              viewBox="0 0 576 512"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M556.686 290.542L410.328 64.829C397.001 44.272 374.417 32 349.917 32H56C25.121 32 0 57.122 0 88v8c0 44.112 35.888 80 80 80h196.042l-18.333 48H144c-48.523 0-88 39.477-88 88 0 30.879 25.121 56 56 56h131.552c2.987 0 5.914.549 8.697 1.631L352 408.418V480h224V355.829c0-23.225-6.679-45.801-19.314-65.287zM528 432H400v-23.582c0-19.948-12.014-37.508-30.604-44.736l-99.751-38.788A71.733 71.733 0 0 0 243.552 320H112c-4.411 0-8-3.589-8-8 0-22.056 17.944-40 40-40h113.709c19.767 0 37.786-12.407 44.84-30.873l24.552-64.281c8.996-23.553-8.428-48.846-33.63-48.846H80c-17.645 0-32-14.355-32-32v-8c0-4.411 3.589-8 8-8h293.917c8.166 0 15.693 4.09 20.137 10.942l146.358 225.715A71.84 71.84 0 0 1 528 355.829V432z"
              />
            </svg>
            Lizard
          </button>
        </div>
        <div
          class="_cL4RU _17_OS  _3W7Om _CWiZp _39_ND _z9EYS button-0-2-17"
        >
          <button
            class="_3lLt8"
          >
            <svg
              class="icon-0-2-19"
              color="#2196f3"
              fill="currentColor"
              height="20"
              size="20"
              stroke="currentColor"
              stroke-width="0"
              style="color: rgb(33, 150, 243);"
              viewBox="0 0 512 512"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M501.03053,116.17605c-19.39059-31.50779-51.24406-35.72849-66.31044-35.01756-14.11325-50.81051-62.0038-54.08-70.73816-54.08a74.03091,74.03091,0,0,0-72.23816,58.916l-4.64648,22.66014-13.68357-53.207c-9.09569-35.37107-46.412-64.05074-89.66-53.07223a73.89749,73.89749,0,0,0-55.121,78.94722,73.68273,73.68273,0,0,0-64.8495,94.42181l24.35933,82.19721c-38.24017-7.54492-62.79677,16.18358-68.11512,21.84764a73.6791,73.6791,0,0,0,3.19921,104.19329l91.36509,85.9765A154.164,154.164,0,0,0,220.62279,512h107.4549A127.30079,127.30079,0,0,0,452.3392,413.86139l57.623-241.96272A73.20274,73.20274,0,0,0,501.03053,116.17605Zm-37.7597,44.60544L405.64788,402.74812a79.46616,79.46616,0,0,1-77.57019,61.25972H220.62279a106.34052,106.34052,0,0,1-73.1366-28.998l-91.369-85.98041C31.34381,325.72669,66.61133,288.131,91.39644,311.5392l51.123,48.10739c5.42577,5.10937,13.48239.71679,13.48239-5.82617a246.79914,246.79914,0,0,0-10.17771-70.1523l-36.01362-121.539c-9.7324-32.88279,39.69916-47.27145,49.38664-14.625l31.3437,105.77923c5.59374,18.90428,33.78119,10.71288,28.9648-8.00781L177.06427,80.23662c-8.50389-33.1035,41.43157-45.64646,49.86515-12.83593l47.32609,184.035c4.42773,17.24218,29.16207,16.5039,32.71089-.80468l31.791-154.9706c6.81054-33.1074,57.51748-24.10741,50.11906,11.96288L360.32764,246.78924c-3.72265,18.10936,23.66793,24.63084,28.05659,6.21679L413.185,148.85962C421.1498,115.512,471.14,127.79713,463.27083,160.78149Z"
              />
            </svg>
            Spock
          </button>
        </div>
        <div
          class="_cL4RU _17_OS  _3W7Om _CWiZp _39_ND _z9EYS button-0-2-17"
        >
          <button
            class="_3lLt8"
          >
            Tactical Mode
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
